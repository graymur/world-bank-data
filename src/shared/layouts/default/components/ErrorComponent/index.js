import React from 'react';
import PropTypes from 'prop-types';
import './error.scss';

const ErrorComponent = ({error}) => {
	if (process.env.NODE_ENV === 'development') {
		console.error(error);
	}

	return (
		<div className='app-error'>
			<h1>Something went wrong:</h1>
			<p>{error.message || error.toString()}</p>
		</div>
	);
};

ErrorComponent.propTypes = {
	error: PropTypes.any
};

export default ErrorComponent;
