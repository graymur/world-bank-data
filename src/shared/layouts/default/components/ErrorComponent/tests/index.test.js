/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import ErrorComponent from '../index';

test('should render an <span> tag', () => {
	const component = renderer.create(<ErrorComponent error='Some error' />);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
