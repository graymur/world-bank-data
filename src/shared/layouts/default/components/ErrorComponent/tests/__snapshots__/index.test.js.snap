// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an <span> tag 1`] = `
<div
  className="app-error"
>
  <h1>
    Something went wrong:
  </h1>
  <p>
    Some error
  </p>
</div>
`;
